pkg/machines/libvirt-common.js

```diff
@@ -144,6 +143,18 @@ export function getElem(xml) {
     return xmlDoc.firstElementChild;
 }
 
+export function getDomainCapMaxVCPU(capsXML) {
+    const domainCapsElem = getElem(capsXML);
+    const vcpuElem = domainCapsElem.getElementsByTagName("vcpu")[0];
```

Comment:
Is it guaranteed that these elements are always present? This could be done a bit more robustly, with returning `null` if the number is unknown, and not offering changing CPU count in that case. Or is that an XML API guarantee?

Reply:
Right - the documentation says these elements *might* exist.

https://libvirt.org/formatdomaincaps.html#elementsCPUAllocation

Reply:
You are right, in case it's missing this would lead to an Oops